components/compositing/compositor.rs

```diff
@@ -1141,14 +1141,41 @@ impl<Window: WindowMethods + ?Sized> IOCompositor<Window> {
             };
             let cursor = (combined_event.cursor.to_f32() / self.scale).to_untyped();
             let cursor = webrender_api::units::WorldPoint::from_untyped(cursor);
-            let mut txn = webrender_api::Transaction::new();
-            txn.scroll(scroll_location, cursor);
+            let mut txn = render_api::Transaction::new();
+
+            let hit = self
+                .webrender_api
+                .hit_test(self.webrender_document, None, cursor);
+            // TODO(bryce): Find the right item (don't just pick the first like I did) and
+            //  actually do this correctly. Note that the results are from front to back
+            if let Some(item) = hit.items.first() {
+                let new_point = match scroll_location {
+                    ScrollLocation::Delta(delta) => delta,
+                    // TODO(bryce): I have no idea what to do here
+                    ScrollLocation::Start => LayoutVector2D::new(0f32, 0f32),
+                    ScrollLocation::End => LayoutVector2D::new(1000f32, 1000f32),
+                };
+                // TODO(bryce): Figure this out. I think the pre-scrolled location in the display
+                //  list needs to be modified but I don't know how to do that yet
+                //println!("{:?}", new_point);
+                //item.point_relative_to_item +
+                //let node_address = UntrustedNodeAddress(item.tag.0 as *const c_void);
+                //let node = unsafe { from_untrusted_node_address(node_address) };
+                txn.scroll_node_with_id(
+                    new_point.to_point(),
```

Comment:
This is where the logic will need to be put in place to store the scroll position. There probably needs to be more scroll logic here is general.